python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -438,6 +436,86 @@ def create_id_map(ids):
     """
     return {id1: i for i, id1 in enumerate(ids)}
 
+def worker_fn(task_queue, res_queue, user_parser):
+    """ The worker function in the worker pool
+
+    Parameters
+    ----------
+    task_queue : Queue
+        The queue that contains all tasks
+    res_queue : Queue
+        The queue that contains the processed data. This is used for
+        communication between the worker processes and the master process.
+    user_parser : callable
+        The user-defined function to read and process the data files.
+    """
+    try:
+        while True:
+            # If the queue is empty, it will raise the Empty exception.
+            i, in_file = task_queue.get_nowait()
+            data = user_parser(in_file)
+            res_queue.put((i, data))
+            gc.collect()
+    except queue.Empty:
+        pass
+
+class WorkerPool:
+    """ A worker process pool
+
+    This worker process pool is specialized to process node/edge data.
+    It creates a set of worker processes, each of which runs a worker function.
+    It first adds all tasks in a queue and each worker gets a task from the queue
+    at a time until the workers process all tasks. It maintains a result queue
+    and each worker adds the processed results in the result queue.
+    To ensure the order of the data, each data file is associated with
+    a number and the processed data are ordered by that number.
+
+    Parameters
+    ----------
+    name : str or tuple of str
+        The name of the worker pool.
+    in_files : list of str
+        The input data files.
+    num_processes : int
+        The number of processes that run in parallel.
+    user_parser : callable
+        The user-defined function to read and process the data files.
+    """
+    def __init__(self, name, in_files, num_processes, user_parser):
+        self.name = name
+        self.processes = []
+        manager = multiprocessing.Manager()
+        self.task_queue = manager.Queue()
```

Comment:
Can we make the size of res_queue tunable? Or 8 is good enough?

Reply:
we probably don't need to tune that much. let's set a default number for users first. otherwise, there will be too many arguments for users to understand.